### PR TITLE
Merge rbtree into sqlite

### PR DESCRIPTION
**Pull request by Lopuiz**
Merged RBtree FIM rework on SQLite FIM rework.

### Test
- [x] Scan-build on Windows and Linux
- [x] Moving and renaming directories are reported
- [x] Compilation without warnings

### CODE REVIEW

File: src/syscheckd/create_db.c [hunk 1]
```diff
@@ -301,17 +301,102 @@ int fim_file(char *file, fim_element *item, whodata_evt *w_evt, int report) {
 }
 
 
-void fim_realtime_event(char *file) {
-    fim_audit_inode_event(file, FIM_REALTIME, NULL);
+void fim_realtime_event(char *file, fim_element *item) {
+
+    struct stat file_stat;
+
+    // If the file exist, generate add or modify events.
```
Comment (vikman90) [inline src/syscheckd/create_db.c]: Please review the English grammar.

File: src/syscheckd/create_db.c [hunk 2]
```diff
@@ -301,17 +301,102 @@ int fim_file(char *file, fim_element *item, whodata_evt *w_evt, int report) {
...
+        fim_checker(pathname, item, w_evt, 1);
+#else
+        fim_audit_inode_event(pathname, mode, w_evt);
+#endif
+
+        return;
+    }
+
+    // Since the file doesn't exist, research if it's directory and have files in DB.
+    fim_tmp_file *file = NULL;
```
Comment (vikman90) [inline src/syscheckd/create_db.c]: `file` is an object that represents a set of files. Please refactor that to `files`.

File: src/syscheckd/fim_db.c [hunk 3]
```diff
@@ -1078,6 +1082,24 @@ void fim_db_remove_path(fdb_t *fim_sql, fim_entry *entry, pthread_mutex_t *mutex
         w_mutex_unlock(mutex);
 }
 
+void fim_db_process_path(fdb_t *fim_sql, fim_entry *entry, pthread_mutex_t *mutex, __attribute__((unused))void *arg) {
+
+    fim_event_mode *mode = (fim_event_mode *) arg;
+    int conf_file = fim_configuration_directory(entry->path, "file");
+
+    if(mode != FIM_REALTIME && syscheck.opts[conf_file] & REALTIME_ACTIVE) {
```
Comment (vikman90) [inline src/syscheckd/fim_db.c]: This part of the code should be programmed using a `switch` statement. It's more efficient and makes more sense.

File: src/syscheckd/run_realtime.c [hunk 4]
```diff
@@ -91,6 +91,7 @@ int realtime_adddir(const char *dir, __attribute__((unused)) int whodata)
 
                 if (!OSHash_Get_ex(syscheck.realtime->dirtb, wdchar)) {
                     if (retval = OSHash_Add_ex(syscheck.realtime->dirtb, wdchar, data), retval == 0) {
+                        os_free(data);
```
Comment (vikman90) [inline src/syscheckd/run_realtime.c]: If this code means that `data` may remain dangling if `OSHash_Add_ex()` returns `0` (error), we should also contemplate that `OSHash_Add_ex()` returned `1` (duplicate entry).

I suggest:

```c
if (retval = OSHash_Add_ex(...), retval != 2) {
   ...
}
```

Comment (Lopuiz) [reply src/syscheckd/run_realtime.c]: I review it and data is free
```C
if (retval = OSHash_Add_ex(syscheck.realtime->dirtb, wdchar, data), retval == 0) {
     os_free(data);
     merror_exit(FIM_CRITICAL_ERROR_OUT_MEM);
 } else if (retval == 1) {
     mdebug2(FIM_REALTIME_HASH_DUP, data);
     os_free(data);
 }
mdebug1(FIM_REALTIME_NEWDIRECTORY, dir);
```

File: src/syscheckd/create_db.c [hunk 5]
```diff
@@ -301,17 +301,102 @@ int fim_file(char *file, fim_element *item, whodata_evt *w_evt, int report) {
...
 
+void fim_process_missing_entry(char * pathname, fim_event_mode mode, whodata_evt * w_evt, fim_element *item) {
+
+    fim_entry *saved_data;
+    int found = 0;
+
+    // Search path in DB.
+    w_mutex_lock(&syscheck.fim_entry_mutex);
+    saved_data = fim_db_get_path(syscheck.database, pathname);
+
```
Comment (0xN3utr0n) [inline src/syscheckd/create_db.c]: Insert  `w_mutex_unlock(&syscheck.fim_entry_mutex);` here to reduce the blocking time.

File: src/syscheckd/fim_db.c [hunk 6]
```diff
@@ -528,6 +528,10 @@ int fim_db_delete_range(fdb_t * fim_sql, fim_tmp_file *file, pthread_mutex_t *mu
             storage, (void *) (int) 0);
 }
 
+int fim_db_process_missing_entry(fdb_t *fim_sql, fim_tmp_file *file, pthread_mutex_t *mutex, int storage, fim_event_mode mode) {
```
Comment (0xN3utr0n) [inline src/syscheckd/fim_db.c]: Add a declaration in fim_db.h explaining what it does.

**Comment by wazuhci** [comment]
The stress test has been launched. The charts will be posted when they are ready.
I will use the following configuration:
### General
- Commit: 75d1046
- Packages: 
  - [manager 3.12.0 0.branchmergerbtreeintosqlite.commit75d1046 rpm](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/rpm/var/wazuh-manager-3.12.0-0.branchmergerbtreeintosqlite.commit75d1046.x86_64.rpm)
  - [agent 3.12.0 0.branchmergerbtreeintosqlite.commit75d1046 rpm](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/rpm/var/wazuh-agent-3.12.0-0.branchmergerbtreeintosqlite.commit75d1046.x86_64.rpm)
  - [manager 3.12.0 0.branchmergerbtreeintosqlite.commit75d1046 deb](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/deb/var/wazuh-manager_3.12.0-0.branchmergerbtreeintosqlite.commit75d1046_amd64.deb)
  - [agent 3.12.0 0.branchmergerbtreeintosqlite.commit75d1046 deb](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/deb/var/wazuh-agent_3.12.0-0.branchmergerbtreeintosqlite.commit75d1046_amd64.deb)
  - [agent 3.12.0 0.branchmergerbtreeintosqlite.commit75d1046 msi](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/windows/wazuh-agent-3.12.0-0.branchmergerbtreeintosqlite.commit75d1046.msi)

### Stress test parameters:
- Manager system: `centos`
- Launch CentOS agent: `true`
- Launch Ubuntu agent: `true`
- Launch Windows agent: `true`
- Test duration: `60`
- Scan frequency: `10`
- Modules to test: 
  - logcollector: `true`
  - syscheck: `true`
  - rootcheck: `true`
  - sca: `true`
  - osquery: `true`
  - syscollector: `true`
  - active-response: `true`
  - azure-logs: `true`
  - cis-cat: `true`
  - docker-listener: `true`
  - open-scap: `true`
  - vulnerability-detector: `false`
- FIM special test: `None`
- FIM sleep: `300`
- FIM max files: `25000`
- SCA sleep: `30`
- SCA max registers: `5000`
- Docker image name: `hello-world`
- Docker sleep: `5`
- Docker jobs: `10`
- Logcollector special test: `None`
- Logcollector max files: `50000`
- Logcollector creation sleep: `300`
- Logcollector write files: `1000`
- Logcollector write sleep: `10`
- Vuln sleep: `60`
- Authd port: `1515`
- Syscollector package: `imagemagick.app`
- Syscollector package version: `7.0.8.48`

Regards!.

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 75d10463b1b027de3fb122597a30b9969a7b3c11
**Parameters:** track-origins=yes leak-check=full keep-stacktraces=alloc-and-free track-fds=yes num-callers=20 show-leak-kinds=definite,indirect
**Coverage:** 

```json
Lines executed:30.68% of 1832

```
**Coverage report detail:** [here](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B262/coverage-report-detail)
## Valgrind result:
```
==15393== Memcheck, a memory error detector
==15393== Copyright (C) 2002-2017, and GNU GPLd, by Julian Seward et al.
==15393== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==15393== Command: /var/ossec/bin/ossec-logcollector -f
==15393== Parent PID: 1
==15393== 
==15393== 
==15393== FILE DESCRIPTORS: 17 open at exit.
==15393== Open file descriptor 15: /var/log/duplicated_file.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 14: /var/log/target_file.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 13: /var/log/label_file.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 12: /var/log/json_file.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 11: /var/log/alias_file.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 10: /var/log/database_file.log-20-02-21.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 9: /var/log/remove_file.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 8: /var/log/single_file.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 7: /var/log/dpkg.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open file descriptor 6: /var/ossec/logs/active-responses.log
==15393==    at 0x621DD19: open (open64.c:47)
==15393==    by 0x619A589: _IO_file_open (fileops.c:189)
==15393==    by 0x619A589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==15393==    by 0x618CEA9: __fopen_internal (iofopen.c:78)
==15393==    by 0x618CEA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==15393==    by 0x176807: reload_file (logcollector.c:903)
==15393==    by 0x174FE3: LogCollectorStart (logcollector.c:380)
==15393==    by 0x17248A: main (main.c:187)
==15393== 
==15393== Open AF_UNIX socket 16: /var/ossec/queue/ossec/logcollector
==15393==    at 0x6230EC7: socket (syscall-template.S:78)
==15393==    by 0x257512: OS_BindUnixDomain (os_net.c:153)
==15393==    by 0x16F5A4: lccom_main (lccom.c:102)
==15393==    by 0x5EF66DA: start_thread (pthread_create.c:463)
==15393==    by 0x622F88E: clone (clone.S:95)
==15393== 
==15393== Open AF_UNIX socket 5: <unknown>
==15393==    at 0x6230EC7: socket (syscall-template.S:78)
==15393==    by 0x2577D9: OS_ConnectUnixDomain (os_net.c:203)
==15393==    by 0x24912E: StartMQ (mq_op.c:36)
==15393==    by 0x172438: main (main.c:182)
==15393== 
==15393== Open file descriptor 4: /dev/urandom
==15393==    at 0x5F00D2B: open (open64.c:47)
==15393==    by 0x25393E: randombytes (randombytes.c:62)
==15393==    by 0x253A70: srandom_init (randombytes.c:82)
==15393==    by 0x171DC6: main (main.c:60)
==15393== 
==15393== Open file descriptor 3: /tmp/dest/Analysis_dynamic_valgrind-v3.12.0-4623.log
==15393==    <inherited from parent>
==15393== 
==15393== Open file descriptor 2:
==15393==    <inherited from parent>
==15393== 
==15393== Open file descriptor 1:
==15393==    <inherited from parent>
==15393== 
==15393== Open file descriptor 0: /dev/null
==15393==    <inherited from parent>
==15393== 
==15393== 
==15393== HEAP SUMMARY:
==15393==     in use at exit: 32,463 bytes in 132 blocks
==15393==   total heap usage: 7,825 allocs, 7,693 frees, 4,603,417 bytes allocated
==15393== 
==15393== LEAK SUMMARY:
==15393==    definitely lost: 0 bytes in 0 blocks
==15393==    indirectly lost: 0 bytes in 0 blocks
==15393==      possibly lost: 1,904 bytes in 7 blocks
==15393==    still reachable: 30,559 bytes in 125 blocks
==15393==         suppressed: 0 bytes in 0 blocks
==15393== Reachable blocks (those to which a pointer was found) are not shown.
==15393== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==15393== 
==15393== For counts of detected and suppressed errors, rerun with: -v
==15393== ERROR SUMMARY: 3 errors from 3 contexts (suppressed: 0 from 0)

```

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 75d10463b1b027de3fb122597a30b9969a7b3c11
**Parameters:** track-origins=yes leak-check=full keep-stacktraces=alloc-and-free track-fds=yes num-callers=20 show-leak-kinds=definite,indirect
**Coverage:** 

```json
Lines executed:26.12% of 2963

```
**Coverage report detail:** [here](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.0/B263/coverage-report-detail)
## Valgrind result:
```
==1752== Memcheck, a memory error detector
==1752== Copyright (C) 2002-2017, and GNU GPLd, by Julian Seward et al.
==1752== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==1752== Command: /var/ossec/bin/ossec-syscheckd -f
==1752== Parent PID: 15965
==1752== 
==1752== 
==1752== FILE DESCRIPTORS: 11 open at exit.
==1752== Open file descriptor 12: /usr/sbin/tcpdump
==1752==    at 0x6222D19: open (open64.c:47)
==1752==    by 0x619F589: _IO_file_open (fileops.c:189)
==1752==    by 0x619F589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==1752==    by 0x6191EA9: __fopen_internal (iofopen.c:78)
==1752==    by 0x6191EA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==1752==    by 0x1A184F: os_string (os_string.c:169)
==1752==    by 0x199C46: check_rc_trojans (check_rc_trojans.c:91)
==1752==    by 0x1976E6: run_rk_check (run_rk_check.c:140)
==1752==    by 0x197D3A: w_rootcheck_thread (run_rk_check.c:317)
==1752==    by 0x5EFB6DA: start_thread (pthread_create.c:463)
==1752==    by 0x623488E: clone (clone.S:95)
==1752== 
==1752== Open file descriptor 10: /var/ossec/etc/rootcheck/rootkit_trojans.txt
==1752==    at 0x6222D19: open (open64.c:47)
==1752==    by 0x619F589: _IO_file_open (fileops.c:189)
==1752==    by 0x619F589: _IO_file_fopen@@GLIBC_2.2.5 (fileops.c:281)
==1752==    by 0x6191EA9: __fopen_internal (iofopen.c:78)
==1752==    by 0x6191EA9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==1752==    by 0x197673: run_rk_check (run_rk_check.c:135)
==1752==    by 0x197D3A: w_rootcheck_thread (run_rk_check.c:317)
==1752==    by 0x5EFB6DA: start_thread (pthread_create.c:463)
==1752==    by 0x623488E: clone (clone.S:95)
==1752== 
==1752== Open file descriptor 9: /dev/urandom
==1752==    at 0x5F05DAE: open (open64.c:47)
==1752==    by 0x27B586: randombytes (randombytes.c:62)
==1752==    by 0x27B71C: os_random (randombytes.c:88)
==1752==    by 0x27348F: OSHash_Create (hash_op.c:54)
==1752==    by 0x17868B: realtime_start (run_realtime.c:35)
==1752==    by 0x17892B: realtime_adddir (run_realtime.c:67)
==1752==    by 0x1797DE: fim_scan (create_db.c:59)
==1752==    by 0x17E619: start_daemon (run_check.c:174)
==1752==    by 0x185EB8: main (main.c:269)
==1752== 
==1752== Open file descriptor 7: /var/ossec/queue/fim/db/fim.db
==1752==    at 0x5F05DAE: open (open64.c:47)
==1752==    by 0x53943A6: posixOpen (sqlite3.c:30223)
==1752==    by 0x5394819: robust_open (sqlite3.c:30516)
==1752==    by 0x5399B83: unixOpen (sqlite3.c:35680)
==1752==    by 0x53892C1: sqlite3OsOpen (sqlite3.c:20291)
==1752==    by 0x53A4FDE: sqlite3PagerOpen (sqlite3.c:51704)
==1752==    by 0x53B2A98: sqlite3BtreeOpen (sqlite3.c:61370)
==1752==    by 0x545F147: openDatabase (sqlite3.c:143418)
==1752==    by 0x545F50B: sqlite3_open_v2 (sqlite3.c:143595)
==1752==    by 0x18F011: fim_db_init (fim_db.c:241)
==1752==    by 0x18080A: fim_initialize (syscheck.c:84)
==1752==    by 0x185D47: main (main.c:245)
==1752== 
==1752== Open AF_UNIX socket 6: <unknown>
==1752==    at 0x6235EC7: socket (syscall-template.S:78)
==1752==    by 0x27FEEB: OS_ConnectUnixDomain (os_net.c:203)
==1752==    by 0x2723B7: StartMQ (mq_op.c:36)
==1752==    by 0x185658: main (main.c:177)
==1752== 
==1752== Open AF_UNIX socket 4: <unknown>
==1752==    at 0x6235EC7: socket (syscall-template.S:78)
==1752==    by 0x27FEEB: OS_ConnectUnixDomain (os_net.c:203)
==1752==    by 0x2723B7: StartMQ (mq_op.c:36)
==1752==    by 0x196B61: rootcheck_connect (rootcheck.c:249)
==1752==    by 0x18563E: main (main.c:173)
==1752== 
==1752== Open AF_UNIX socket 5: /var/ossec/queue/ossec/syscheck
==1752==    at 0x6235EC7: socket (syscall-template.S:78)
==1752==    by 0x27FC24: OS_BindUnixDomain (os_net.c:153)
==1752==    by 0x1844C7: syscom_main (syscom.c:125)
==1752==    by 0x5EFB6DA: start_thread (pthread_create.c:463)
==1752==    by 0x623488E: clone (clone.S:95)
==1752== 
==1752== Open file descriptor 3: /tmp/dest/Analysis_dynamic_valgrind-v3.12.0-4623.log
==1752==    <inherited from parent>
==1752== 
==1752== Open file descriptor 2:
==1752==    <inherited from parent>
==1752== 
==1752== Open file descriptor 1:
==1752==    <inherited from parent>
==1752== 
==1752== Open file descriptor 0: /dev/null
==1752==    <inherited from parent>
==1752== 
==1752== 
==1752== HEAP SUMMARY:
==1752==     in use at exit: 459,284 bytes in 738 blocks
==1752==   total heap usage: 3,473,458 allocs, 3,472,720 frees, 466,062,375 bytes allocated
==1752== 
==1752== LEAK SUMMARY:
==1752==    definitely lost: 0 bytes in 0 blocks
==1752==    indirectly lost: 0 bytes in 0 blocks
==1752==      possibly lost: 5,464 bytes in 6 blocks
==1752==    still reachable: 453,820 bytes in 732 blocks
==1752==                       of which reachable via heuristic:
==1752==                         length64           : 263,024 bytes in 307 blocks
==1752==         suppressed: 0 bytes in 0 blocks
==1752== Reachable blocks (those to which a pointer was found) are not shown.
==1752== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==1752== 
==1752== For counts of detected and suppressed errors, rerun with: -v
==1752== ERROR SUMMARY: 6 errors from 6 contexts (suppressed: 0 from 0)

```

**Comment by wazuhci** [comment]
## Stress/Leaks test results
### Windows Agent ID0
- #### CPU
    [Windows Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-winagent-ID0_B169_Test_stress_windows_Agent-warehouse-pullrequests_CPU(%25).svg)
- #### File Descriptors
    [Windows Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-winagent-ID0_B169_Test_stress_windows_Agent-warehouse-pullrequests_FD.svg)
- #### Resident Set Size memory
    [Windows Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-winagent-ID0_B169_Test_stress_windows_Agent-warehouse-pullrequests_RSS(KB).svg)
### CentOS Manager ID0
- #### CPU
    [CentOS Manager](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-manager-ID0_B169_Test_stress_Manager-warehouse-pullrequests_CPU(%25).svg)
- #### File Descriptors
    [CentOS Manager](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-manager-ID0_B169_Test_stress_Manager-warehouse-pullrequests_FD.svg)
- #### Resident Set Size memory
    [CentOS Manager](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-manager-ID0_B169_Test_stress_Manager-warehouse-pullrequests_RSS(KB).svg)
### CentOS Agent ID1
- #### CPU
    [CentOS Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-agent-ID1_B169_Test_stress_centos_Agent-warehouse-pullrequests_CPU(%25).svg)
- #### File Descriptors
    [CentOS Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-agent-ID1_B169_Test_stress_centos_Agent-warehouse-pullrequests_FD.svg)
- #### Resident Set Size memory
    [CentOS Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-agent-ID1_B169_Test_stress_centos_Agent-warehouse-pullrequests_RSS(KB).svg)
### Ubuntu Agent ID0
- #### CPU
    [Ubuntu Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-agent-ID0_B169_Test_stress_ubuntu_Agent-warehouse-pullrequests_CPU(%25).svg)
- #### File Descriptors
    [Ubuntu Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-agent-ID0_B169_Test_stress_ubuntu_Agent-warehouse-pullrequests_FD.svg)
- #### Resident Set Size memory
    [Ubuntu Agent](https://ci.wazuh.com/data/Test_stress/warehouse-pullrequests/3.12.0/B169/monitor-agent-ID0_B169_Test_stress_ubuntu_Agent-warehouse-pullrequests_RSS(KB).svg)

**Pull request merged**